api/src/constants.ts

```diff
@@ -38,7 +38,7 @@ export const SYSTEM_ASSET_ALLOW_LIST: TransformationParams[] = [
 	},
 ];
 
-export const ASSET_TRANSFORM_QUERY_KEYS: Array<keyof TransformationParams> = [
+export const ASSET_TRANSFORM_QUERY_KEYS = [
```

Comment:
I removed it because it doesnt let us know whats actually inside of the array which then leads to wrong type inference down the line. I've ran into that specifically, thats why I removed it.

Let's say the `TransformationParams` changes in the future or we forget one key like so:

```ts
export const ASSET_TRANSFORM_QUERY_KEYS: Array<keyof TransformationParams> = [
	'key',
	'transforms',
	'width',
	'height',
	'format',
	'fit',
	'quality',
	'withoutEnlargement',
	'focal_point_x',
	// 'focal_point_y',
];
```

Now we dont know whether or not that key will ever be picked in the assets controller. The type doesnt let us know that we will never actually pick this key:

![image](https://github.com/directus/directus/assets/14810858/60492ead-a57c-41fc-81ba-ad194fddddfc)

By changing it to readonly we get the correct keys and avoid this point of failure.

Can we preserve the correct array items in the initialization but still mark it as readonly? :thinking: cc @br41nslug

Reply:
Talked to @br41nslug and he came up with the big brain solution of 

` as const satisfies (keyof TransformationParams)[]`

which does exactly what I wanted!!! :heart_eyes: 

With that we get the correct items, we error on wrong items and we do actually get the content of said array for the picking! win, win, win!

Reply:
Discussed with Daniel and this looks like the best of both worlds:
```ts
export const ASSET_TRANSFORM_QUERY_KEYS = [
	'key',
	'transforms',
	'width',
	'height',
	'format',
	'fit',
	'quality',
	'withoutEnlargement',
	'focal_point_x',
	'focal_point_y',
]  as const satisfies (keyof TransformationParams)[];
```

Reply:
Right, this is the perfect solution 👍